pghoard/common.py

```diff
@@ -231,7 +231,10 @@ def increase_pipe_capacity(*pipes):
     try:
         with open('/proc/sys/fs/pipe-max-size', 'r') as f:
             pipe_max_size = int(f.read())
-            for pipe in pipes:
-                fcntl.fcntl(pipe, 1031, pipe_max_size)
     except FileNotFoundError:
+        return
+    try:
+        for pipe in pipes:
+            fcntl.fcntl(pipe, 1031, pipe_max_size)  # F_SETPIPE_SZ
+    except PermissionError:
```

Comment:
Any idea what's the more specific scenario in which this happens? Having a comment here about that would be nice. While nothing will break not being able to increase the capacity will at least increase context switching quite a bit, slowing things down so I'd be interesting in knowing how common this might be.

Reply:
It happens in my unit tests (systemd-nspawn container, non-root user, latest Fedora 30). There's 3 different reasons for the error: # of kernel pages would be more than number of pages allocated to pipe, the pipe_max_size check (that we sort of do above), and finally not having CAP_SYS_RESOURCE capability.

I am bit leery of commenting about kernel internals in code though, as those do change.

Reply:
My comment was bit incorrect, but anyway as nonroot there's few cases where it occurs. There's soft and hard limits of number of per-user buffers.